### PR TITLE
Assaf/gp/crm

### PR DESCRIPTION
**Pull request by Faskoman**

### CODE REVIEW

File: api/src/index.ts [hunk 1]
```diff
@@ -62,6 +63,40 @@ app.post("/registerlead", async (req, res) => {
...
+      priority,
+      priority,
+      stage,
+      stage,
+    ]);
+
+    res.status(200).json(result);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ error: "something went wrong" });
```
Comment (yizhak148) [inline api/src/index.ts:96]: You need to be more specific when defining the error.

File: app/src/LeadsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
+import axios from "axios";
+import "./LeadsPage.scss";
+import { useState, useEffect, PropsWithChildren } from "react";
+
+const server = axios.create({
```
Comment (yizhak148) [inline app/src/LeadsPage.tsx:5]: delete that part, it's not necessary.

File: app/src/LeadsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+export function LeadsPage() {
+  const [leads, setLeads] = useState<Lead[]>([]);
+  const [search, setSearch] = useState<string>("");
+  const [priority, setPriority] = useState<string>("");
+  const [stage, setStage] = useState<string>("");
+
+  useEffect(() => {
+    const fetchLeads = async () => {
+      try {
+        const response = await server.get(
```
Comment (yizhak148) [inline app/src/LeadsPage.tsx:41]: fix that function according to the change at line 5.

File: app/src/LeadsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  owner?: string;
+  firstName: string;
+  lastName: string;
+  phoneNumber: string;
+  email: string;
+  jobTitle: string;
+  companyName: string;
+  sector: string;
+  employeeCount?: number;
+  address?: string;
```
Comment (yizhak148) [inline app/src/LeadsPage.tsx:29]: there is alreday a file that defined a "lead" type...

File: app/src/LeadsPage.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            <th>Priority</th>
+            <th>Stage</th>
+            <th>Created at</th>
+          </tr>
+        </thead>
+        <tbody>
+          {leads.map((lead, index) => (
+            <tr key={index} className="leadsTable__leadInfo">
+              <td className="leadsTable__leadInfo__name">
+                <a href={`leads/${lead.id}`}>
```
Comment (yizhak148) [inline app/src/LeadsPage.tsx:98]: change the <a> tag to <Link> tag.

**Review by yizhak148 — COMMENTED** [review]
go over the comment and make sure you fix them all.

File: app/src/LeadsPage.tsx (hunk 4 above)
Comment (Faskoman) [reply app/src/LeadsPage.tsx:29]: lilach and me have to decide on a way to pull out the leads into a nested object.

File: app/src/LeadsPage.tsx (hunk 5 above)
Comment (Faskoman) [reply app/src/LeadsPage.tsx:98]: changed it to Link to

**Pull request merged**